crates/core/src/logstore/mod.rs

```diff
@@ -338,6 +338,14 @@ pub trait LogStore: Send + Sync + AsAny {
         self.to_uri(&Path::from(""))
     }
 
+    fn table_root_url(&self) -> Url {
```

Comment:
I'm still digesting this whole pull request, but please add some rustdoc for why this function exists. Bonus points in adding a unit test :kissing_smiling_eyes: